speech/snippets/multichannel.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2016 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// Command caption sends multichannel audio data to the Google Speech API
+// and prints its transcript.
+// go run multichannel.go ../testdata/commercial_stereo.wav
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+
+	speech "cloud.google.com/go/speech/apiv1p1beta1"
+	speechpb "google.golang.org/genproto/googleapis/cloud/speech/v1p1beta1"
+)
+
+const usage = `Usage: multichannel <audiofile>
+
+Audio file must be a 16-bit signed stereo (2-channel) with a sample rate of 44100.
+
+The path to the audio file may be a GCS URI (gs://...).
+`
+
+func main() {
+	flag.Parse()
+	if len(os.Args) < 2 {
+		fmt.Fprintln(os.Stderr, usage)
+		os.Exit(2)
+	}
+
+	var runFunc func(io.Writer, string) error
+
+	path := os.Args[1]
+	if strings.Contains(path, "://") {
+		runFunc = recognizeMultichannelGCS
+	} else {
+		runFunc = recognizeMultichannel
+	}
+
+	// Perform the request.
+	if err := runFunc(os.Stdout, os.Args[1]); err != nil {
+		log.Fatal(err)
+	}
+}
+
+// [START speech_transcribe_sync_gcs]
+
+func recognizeMultichannelGCS(w io.Writer, gcsURI string) error {
+	ctx := context.Background()
+
+	client, err := speech.NewClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Send the request with the URI (gs://...)
+	// and sample rate information to be transcripted.
+	resp, err := client.Recognize(ctx, &speechpb.RecognizeRequest{
+		Config: &speechpb.RecognitionConfig{
+			Encoding:        speechpb.RecognitionConfig_LINEAR16,
+			SampleRateHertz: 44100,
+			LanguageCode:    "en-US",
+      AudioChannelCount:  2,
+      EnableSeparateRecognitionPerChannel: true,
+		},
+		Audio: &speechpb.RecognitionAudio{
+			AudioSource: &speechpb.RecognitionAudio_Uri{Uri: gcsURI},
+		},
+	})
+
+	// Print the results.
+	for _, result := range resp.Results {
+		for _, alt := range result.Alternatives {
+			fmt.Fprintf(w, "Channel %v: \"%v\" (confidence=%3f)\n", result.ChannelTag, alt.Transcript, alt.Confidence)
```

Comment:
Use `%q` to automatically wrap in quotes.

```suggestion
			fmt.Fprintf(w, "Channel %v: %q (confidence=%3f)\n", result.ChannelTag, alt.Transcript, alt.Confidence)
```

Reply:
chose to remove quotes from transcript instead